drop-transfer/src/ws/client/v3.rs

```diff
@@ -152,7 +152,10 @@ impl HandlerLoop<'_> {
         limit: u64,
     ) -> anyhow::Result<()> {
         let f = || {
-            let xfile = self.xfer.file(&file_id).context("File not found")?;
+            let xfile = self
+                .xfer
+                .file_by_subpath(&file_id)
```

Comment:
Could you clarify if I understand it correctly as it's a lot to sift through:

Transfer now holds the files as a flat list instead of a tree hierarchy and thus has `file_by_subpath()`. This sounds a bit weird as it only makes sense for old versions as new ones will use the hash instead, but we have a FileId conversion to a FileSubpath so as both are strings it just works.

Is it correct that the old versions will pass the glued directory as an ID and also will expect the same while this version will pass and expect a hash instead? This way we're actually breaking the compatibility between this release and others with no indication as it will just respond that such a file was not found.

If what's above is true we then should:
- introduce additional meta-data for `file_by_subpath` so it would resolve either file IDs or hashes

Optionally introduce v4 to not complexify v3 implementation.

Reply:
You're somewhat right. The files as indexed by hash. However old protocols v1..=v3 operate on subpaths, so for this particular `client/v3.rs` implementation we still need to query the files by subpath.
In the next PR I'm going to make a v4 protocol, operating on hashes exclusively. I didn't include it in this PR because it would be humongous.
So this PR changes the subpaths -> hashes in the internals, API layer, and events. The next one will make use of it for the v4 protocol to properly fix the name-clash bugs